packages/ui5-middleware-onelogin/src/index.ts

```diff
@@ -0,0 +1,70 @@
+import { serialize } from 'cookie';
+import dotenv from 'dotenv';
+import cookieGetter from './cookieGetter';
+import {Options} from "./types";
+const log = require('@ui5/logger').getLogger('server:custommiddleware:onelogin');
+dotenv.config();
+
+// interface cookieFace {
+// name: string,
+// key: string
+// }
+/**
+ * Custom UI5 Server middleware example
+ *
+ * @param {Object} parameters Parameters
+ * @param {Object} parameters.resources Resource collections
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.
+ *                                        all Reader or Collection to read resources of the
+ *                                        root project and its dependencies
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.
+ *                                        rootProject Reader or Collection to read resources of
+ *                                        the project the server is started in
+ * @param {module:@ui5/fs.AbstractReader} parameters.resources.dependencies
+ *                                        Reader or Collection to read resources of
+ *                                        the projects dependencies
+ * @param {Object} parameters.options Options
+ * @param {string} [parameters.options.configuration] Custom server middleware configuration
+ *                                                      if given in ui5.yaml
+ * @returns {function} Middleware function to use
+ */
+module.exports = function ({ options }: {options: Options}) {
+  // eslint-disable-next-line func-names
+  return async function (req: any, res: any, next: any) {
+    let cookies = [];
+    if (!process.env.UI5_MIDDLEWARE_ONELOGIN_LOGIN_URL) {
+      next();
+    } else if (!process.env.cookie) {
+      log.info('Fetching cookie, hang on!');
+      const cookieObj = await new cookieGetter().getCookie(options);
+      cookies = JSON.parse(cookieObj);
+      process.env.cookie = cookieObj;
```

Comment:
why do you use the process.env to store the cookie and not locally in a variable? do you want to share the cookie with some other module?

Reply:
I stuffed up the scope in a previous version and it was undefined. I've added to global variable and it seems fine now. Changed

Reply:
Thx, good to know.